src/util/Web.re

```diff
@@ -12,3 +16,115 @@ let div_if = (p, ats, ns) => p ? div(~attrs=[ats], ns) : div_empty;
 let span_if = (p, ats, ns) => p ? span(~attrs=[ats], ns) : span([]);
 
 let unless = (p, a) => p ? Effect.Many([]) : a;
+
```

Comment:
hmm, i forgot about Widgets. maybe we should just combine these modules? do you see a distinction between them? right now i can't just move things to Widgets as it's in web (renegotiating what goes where is likely going to be an ongoing process now that the core/web division is changing)

Reply:
To me web is just like wrappers around html things and widgets has more hazel-specific logic and styles etc - maybe we move widgets to util/Widgets.re if it can't be in web? (I still find it a little concerning that we need an html definition of a slider in core)

Reply:
I had some similar issues on my livelits branch. I struggle to see how the projectors can be web (platform) agnostic. So unless they're entirely configured in web I don't know how you'd excise the dependency from core.

Reply:
@Negabinary i started to move widgets but looking at it the current arrangement makes sense to me. the range and TextArea components i defined in web are intended to be completely independent from hazel logic. all of the textarea logic is just to get and set text area ui state; it could all be in vdom in-principle. otoh current widgets are all related to hazel UI chrome outside the syntax-editor as such. so these modules are disjoint along two different dimensions. admittedly thats not well-reflected in the naming

Reply:
@7h3kk1d contemplating this resulted in a tweet: https://twitter.com/disconcision/status/1818812875411968085

Reply:
(via the tangent "what does it mean to be truly view independent?" => "let's take string out of Core" => "let's string out")